web_src/js/components/RepoContributors.vue

```diff
@@ -0,0 +1,395 @@
+    },
+  },
+  data: () => ({
+    isLoading: false,
+    errorText: '',
+    totalStats: {},
+    sortedContributors: {},
+    repoLink: pageData.repoLink || [],
+    type: pageData.contributionType,
+    contributorsStats: [],
+    dateFrom: null,
+    dateUntil: null,
+    startDate: null,
+    endDate: null,
+  }),
+  mounted() {
+    this.fetchGraphData();
+
+    $('.ui.dropdown').dropdown({
+      onChange: (val) => {
+        this.dateFrom = this.startDate;
+        this.dateUntil = this.endDate;
+        this.type = val;
+        this.sortContributors();
+      }
+    });
+  },
+  methods: {
+    sortContributors() {
+      const contributors = this.filterContributorWeeksByDateRange();
+      const criteria = `total_${this.type}`;
+      this.sortedContributors = Object.values(contributors)
+        .filter((contributor) => contributor[criteria] !== 0)
+        .sort((a, b) => a[criteria] > b[criteria] ? -1 : a[criteria] === b[criteria] ? 0 : 1)
+        .slice(0, 100);
+    },
+
+    async fetchGraphData() {
+      this.isLoading = true;
+      try {
+        let response;
+        do {
+          response = await GET(`${this.repoLink}/activity/contributors/data`);
+          if (response.status === 202) {
+            await new Promise(resolve => setTimeout(resolve, 1000)); // wait for 1 second before retrying
+          }
+        } while (response.status === 202);
+        if (response.ok) {
+          const data = await response.json();
+          const {total, ...rest} = data;
+          this.contributorsStats = rest;
+          this.dateFrom = new Date(total.weeks[0].week);
+          this.dateUntil = new Date();
+          this.startDate = this.dateFrom;
+          this.endDate = this.dateUntil;
+          this.sortContributors();
+          this.totalStats = total;
+          this.errorText = '';
+        } else {
+          this.errorText = response.statusText;
+        }
+      } catch (err) {
+        this.errorText = err.message;
+      } finally {
+        this.isLoading = false;
+      }
+    },
+
+    filterContributorWeeksByDateRange() {
+      const filteredData = {};
+      const data = this.contributorsStats;
+      for (const key of Object.keys(data)) {
+        const user = data[key];
+        user['total_commits'] = 0;
+        user['total_additions'] = 0;
+        user['total_deletions'] = 0;
+        user['max_contribution_type'] = 0;
+        const filteredWeeks = user.weeks.filter((week) => {
+          const weekDate = new Date(week.week);
+          if (weekDate >= this.dateFrom && weekDate <= this.dateUntil) {
+            user['total_commits'] += week.commits;
+            user['total_additions'] += week.additions;
+            user['total_deletions'] += week.deletions;
+            if (week[this.type] > user['max_contribution_type']) {
+              user['max_contribution_type'] = week[this.type];
+            }
+            return true;
+          }
+          return false;
+        });
+
```

Comment:
```suggestion
        user['max_contribution_type'] += 1;
```
To avoid someone misunderstand this again, maybe it is better to set `max_contribution_type += 1` here.

After:
![image](https://github.com/go-gitea/gitea/assets/18380374/aeb46246-746b-436a-8599-e80d17d1f384)

Reply:
@sahinakkaya 
This suggestion is required. Mentioned in https://github.com/sahinakkaya/gitea/pull/3#discussion_r1396495722.

Reply:
No, I mean add it as a comment in the code itself so someone reading the code can understand why that line is necessary @yp05327